octopusdeploy_framework/resource_tag.go

```diff
@@ -146,15 +147,20 @@ func (t *tagTypeResource) Update(ctx context.Context, req resource.UpdateRequest
 		}
 
 		tag := schemas.MapFromStateToTag(data)
+		if tag.ID == "" {
+			tag.ID = destinationTagSet.GetID() + "/" + strings.Split(state.ID.ValueString(), "/")[1]
+		}
 
 		// check to see that the tag is not applied to a tenant
 		isUsed, err := isTagUsedByTenants(ctx, t.Client, sourceTagSetSpaceID, tag)
 		if err != nil {
+			data.ID = types.StringValue("")
```

Comment:
These existed on the old provider but were dropped in the migration, just putting them back.